packages/block-library/src/column/edit.native.js

```diff
@@ -41,69 +38,11 @@ function ColumnEdit( {
 	const minWidth = Math.min( containerWidth, containerMaxWidth );
 	const columnBaseWidth = minWidth / Math.max( 1, columnsInRow );
 
-	const applyBlockStyle = ( placeholder = false ) => {
-		const pullWidths = ( names ) =>
-			names.map(
-				( name ) =>
-					( styles[ `column-${ name }-margin` ] || {} ).width || 0
-			);
-
+	const applyBlockStyle = () => {
 		let width = columnBaseWidth;
-		const names = [
-			'selected',
-			'parent-selected',
-			'descendant-selected',
-			'placeholder-selected',
-			'dashed-border',
-		];
-		const widths = pullWidths( names );
-		const [
-			emptyColumnSelected,
-			parentSelected,
-			columnSelected,
-			placeholderParentSelected,
-			dashedBorderWidth,
-		] = widths;
-
-		switch ( true ) {
-			case isSelected:
-				width = columnBaseWidth;
-				width -= ! hasChildren
-					? emptyColumnSelected
-					: columnSelected + dashedBorderWidth;
-				break;
-
-			case isParentSelected:
-				width -= placeholder
-					? placeholderParentSelected
-					: parentSelected;
-				break;
-
-			case isDescendantSelected:
-				width = columnBaseWidth;
-				break;
-
-			case isDescendantOfParentSelected:
-				width = columnBaseWidth;
-				if ( ! hasChildren ) width -= emptyColumnSelected;
-				break;
-
-			case isAncestorSelected:
-				width = columnBaseWidth;
-				if ( ! hasChildren ) width -= parentSelected;
-				break;
-
-			case placeholder:
-				width -=
-					columnsInRow === 1
-						? parentSelected
-						: placeholderParentSelected;
-				width -=
-					columnSelected +
-					( columnsInRow === 1 ? parentSelected : dashedBorderWidth );
-				break;
-
-			default:
+		if ( columnsInRow > 1 ) {
+			const margins = columnsInRow * 2 * styles.columnMargin.marginLeft;
+			width = ( minWidth - margins ) / Math.max( 1, columnsInRow );
```

Comment:
I have checked and this also works for more than 2 `columnsInRow`.

However we need to implement ideal version of FloatingToolbar because it breaks the layout when there is no enough space (it has minWIdth and the Column container is stretch to this sizes - see below screen). It's not something we need to change here just as reminder)

**Unselected Columns layout**
<image src="https://user-images.githubusercontent.com/21242757/77320124-47385580-6d10-11ea-966e-d100672ade03.png" width="400" />

**Selected Column with broken Columns layout**
<image src="https://user-images.githubusercontent.com/21242757/77320131-4a334600-6d10-11ea-8d29-cc7debfaf28c.png" width="400" />

**Selected Column with proper Columns layout after disable FloatingToolbar**
<image src="https://user-images.githubusercontent.com/21242757/77320134-4acbdc80-6d10-11ea-928a-2b9a2ffc5c08.png" width="400" />

Reply:
Yes it's known issue and exists also in buttons block